contracts/TopShot.cdc

```diff
@@ -1324,6 +1412,185 @@ pub contract TopShot: NonFungibleToken {
+        //
+        // Returns: the ID of the new Subedition object
+        //
+        pub fun createSubedition(name:String, metadata:{String:String}): UInt32 {
+
+            let newID = self.nextSubeditionID
+
+            var newSubedition = Subedition(subeditionID: newID, name: name, metadata: metadata)
+
+            self.nextSubeditionID = self.nextSubeditionID + UInt32(1)
+
+            emit SubeditionCreated(id: newID, name: name, metadata: metadata)
+
+            self.subeditionDatas[newID] = newSubedition
+
+            emit SubeditionCreated(id: newID, name: name, metadata: metadata)
+
+            return newID
+        }
+
+        // getMomentsSubedition function that return's wich Subedition the Moment belongs to
+        //
+        // Parameters: nftID: The ID of the NFT
+        //
+        // returns: UInt32? Subedition's ID if exists
+        //
+        pub fun getMomentsSubedition( nftID: UInt64):UInt32? {
+           return self.momentsSubedition[nftID]
+        }
+
+        // getNumberMintedPerSubedition function that return's
+        // the number of Moments that have been minted for this subedition
+        // to use as this Moment's serial number
+        //
+        // Parameters: setID: The ID of the Set Moment will be minted from
+        //             playID: The ID of the Play Moment will be minted from
+        //             subeditionID: The ID of the Subedition using which moment will be minted
+        //
+        // returns: UInt32 Number of Moments, already minted for this Subedition
+        //
+        pub fun getNumberMintedPerSubedition(setID: UInt32, playID: UInt32, subeditionID: UInt32): UInt32 {
+           let setPlaySubedition = setID.toString().concat(playID.toString()).concat(subeditionID.toString())
+           if !self.numberMintedPerSubedition.containsKey(setPlaySubedition) {
+                self.numberMintedPerSubedition.insert(key: setPlaySubedition,UInt32(0))
+                return UInt32(0)
+           }
+           return self.numberMintedPerSubedition[setPlaySubedition]!
+        }
+
+        // addToNumberMintedPerSubedition function that increments 1 to the
+        // number of Moments that have been minted for this subedition
+        //
+        // Parameters: setID: The ID of the Set Moment will be minted from
+        //             playID: The ID of the Play Moment will be minted from
+        //             subeditionID: The ID of the Subedition using which moment will be minted
+        //
+        //
+        pub fun addToNumberMintedPerSubedition(setID: UInt32, playID: UInt32, subeditionID: UInt32) {
+           let setPlaySubedition = setID.toString().concat(playID.toString()).concat(subeditionID.toString())
+
+           if self.numberMintedPerSubedition.containsKey(setPlaySubedition) {
+              self.numberMintedPerSubedition[setPlaySubedition]!= self.numberMintedPerSubedition[setPlaySubedition]! + UInt32(1)
```

Comment:
Is this supposed to be `!=` ?

Reply:
Oh, just realized it is unwrapping the optional. Maybe a space after the ! would help